src/graph/backend/dnnl/passes/utils.cpp

```diff
@@ -645,6 +646,32 @@ bool inverse_mul_scales(std::shared_ptr<op_t> &scale_op) {
     return true;
 }
 
+bool check_broadcast_requirement_for_inputs_pair(
+        const std::shared_ptr<op_t> &op, size_t index1, size_t index2) {
+    auto in_vals = op->get_input_values();
+
+    const dims input1_dims
+            = logical_tensor_wrapper_t(in_vals[index1]->get_logical_tensor())
+                      .vdims();
+    const dims input2_dims
+            = logical_tensor_wrapper_t(in_vals[index2]->get_logical_tensor())
+                      .vdims();
+
+    const size_t input1_ndims
+            = logical_tensor_wrapper_t(in_vals[index1]->get_logical_tensor())
+                      .ndims();
+    const size_t input2_ndims
+            = logical_tensor_wrapper_t(in_vals[index2]->get_logical_tensor())
+                      .ndims();
+
+    if (input1_ndims != input2_ndims) { return true; }
+
+    for (size_t i = 0; i < input1_ndims; i++) {
+        if (input1_dims[i] != input2_dims[i]) { return true; }
+    }
```

Comment:
Since you already checked the length of the two vectors, I think you can simplify it with:

```suggestion
    if (input1_dims != input2_dims) return true;
```

Reply:
Fixed.